pkg/chart/v2/loader/archive.go

```diff
@@ -233,3 +241,18 @@ func LoadArchive(in io.Reader) (*chart.Chart, error) {
 
 	return LoadFiles(files)
 }
+
+// ParseEnvSettings reads environment variables and updates the size limits
+func parseEnvSettings() {
+	if chartSizeStr := os.Getenv("HELM_MAX_DECOMPRESSED_CHART_SIZE"); chartSizeStr != "" {
+		if size, err := strconv.ParseInt(chartSizeStr, 10, 64); err == nil {
+			MaxDecompressedChartSize = size
+		}
+	}
+
+	if fileSizeStr := os.Getenv("HELM_MAX_DECOMPRESSED_FILE_SIZE"); fileSizeStr != "" {
+		if size, err := strconv.ParseInt(fileSizeStr, 10, 64); err == nil {
+			MaxDecompressedFileSize = size
+		}
+	}
+}
```

Comment:
Helm is divided into an SDK and a CLI/Client. The SDK is used by a wide array of applications (e.g., Flux - a sister project in the CNCF). SDK users have a method to change this value via public package level variables.

To expose this in the CLI/Client, the code for parsing and using these variables should be there and then set the package level variables. This is to have a clean separation.

The Environment Variables should be set in _pkg/cmd/root.go_. This file also includes the right place to document them alongside the other environment variables.

Reply:
Thanks @mattfarina , I will do the change.